islands/formsNossasLojas.tsx

```diff
@@ -0,0 +1,158 @@
+import Icon from "site/components/ui/Icon.tsx";
+
+import { useState } from "preact/compat";
+import { zoom } from "site/islands/Map.tsx";
+import { ItemWithWhatsapp } from "site/sections/NossasLojas/types.ts";
+
+interface StoreInfo {
+  name: string;
+  id: string;
+  address: {
+    postalCode: string;
+    city: string;
+    state: string;
+    neighborhood: string;
+    street: string;
+    number: string;
+    location: {
+      latitude: number;
+      longitude: number;
+    };
+  };
+  businessHours: Array<{
+    dayOfWeek: string;
+    openingTime: string;
+    closingTime: string;
+  }>;
+}
+
+interface Stores {
+  stores: ItemWithWhatsapp[];
+}
+
+const days = [
+    "Domingo",
+    "Segunda-feira",
+    "Terça-feira",
+    "Quarta-feira",
+    "Quinta-feira",
+    "Sexta-feira",
+    "Sábado"
+]
+
+export default function FormsNossasLojas({ stores }: Stores) {
+  const [showFiltered, setShowFiltered] = useState(false);
+  const [queryInput, setQueryInput] = useState("");
+  const [filtered, setFiltered] = useState<ItemWithWhatsapp[]>([]);
+  const [IsCicked, setIsClicked] = useState<number | null>(null);
+
+  const handleClick = (index: number) => {
+    setIsClicked(index);
+  };
+
+  const handleFilter = () => {
+    const result = stores.filter((store) => {
+      return Object.values(store).some((value) =>
+        value.toString().toLowerCase().includes(queryInput.toLowerCase())
+      );
+    });
+    setShowFiltered(!showFiltered);
+    setFiltered(result);
+  };
+
+  const content = showFiltered ? filtered : stores 
+  
+  return (
+    <div class="mt-[30px] mobile:mb-[45px]">
+      <section>
+        <p class="font-bold text-[14px] text-[#676767]">
+          Busque por cidade, CEP, endereço ou loja:
+        </p>
+        <form class="flex justify-between mt-[10px]" action="">
+          <input
+            class="w-[64%] h-[44px] border-[1px] border-[#F7E0BF] rounded-[8px] py-[15px] pl-[20px]"
+            value={queryInput}
+            onChange={(e: React.TargetedEvent<HTMLInputElement>) =>
+              setQueryInput(e.currentTarget.value)}
+            type="text"
+            placeholder="Buscar"
+          />
+          <button
+            class="w-[34%] h-[44px] font-bold text-[#ffffff] bg-[#FF8300] rounded-[8px]"
+            type="button"
+            onClick={handleFilter}
+          >
+            Buscar
```

Comment:
é essencial usar o submit para formulários.
```suggestion
            type="submit"
          >
            Buscar
```

No form:

```diff
- <form class="flex justify-between mt-[10px]" action="">
+ <form class="flex justify-between mt-[10px]" onSubmit={handleFilter}>
```